core/execution/amm/engine.go

```diff
@@ -754,7 +754,17 @@ func (e *Engine) StopPool(
 	return nil
 }
 
-func (e *Engine) MarketClosing() error { return errors.New("unimplemented") }
+// MarketClosing stops all AMM's and returns subaccount balances back to the owning party.
+func (e *Engine) MarketClosing(ctx context.Context) error {
+	for _, p := range e.poolsCpy {
+		if _, err := e.releaseSubAccounts(ctx, p); err != nil {
+			return err
+		}
+		p.status = types.AMMPoolStatusStopped
+		e.remove(ctx, p.party)
```

Comment:
Not sure, but are we sending the events for these status updates?

Reply:
Yup, inside `e.remove()` there is a call to `e.sendUpdate()` that should be shooting out an event with the updated status.